drivers/flash/Kconfig.mcux

```diff
@@ -40,26 +51,23 @@ config FLASH_MCUX_FLEXSPI_NOR
 	depends on DT_HAS_NXP_IMX_FLEXSPI_NOR_ENABLED
 	select FLASH_HAS_PAGE_LAYOUT
 	select FLASH_HAS_DRIVER_ENABLED
-	select MEMC
```

Comment:
This deviates from the pattern we generally use, where devices on a bus `select` the Kconfig of the bus they are on. For example, see sensor drivers that `select I2C` since the sensor driver is on a I2C bus: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/bma280/Kconfig#L10

In my view, the Flash device is on the FlexSPI bus, which is why we `select MEMC` to enable the FlexSPI MEMC driver when using this flash device

Reply:
The logic I have proposed here is "if the flash device is on the flexspi bus and enabled, turn memc on". Please explain how this is different from what you're saying.

Reply:
To be clear, I am not depending on `MEMC` I am depending on `MEMC_MCUX_FLEXSPI` because of the case where flexspi is status okay, memc is on, but the flash controller is not the flexspi because the flexspi is used for something else and/or the MEMC kconfig is turned on for a different reason. This is as you pointed out in #62788. This PR is a more correct way to handle these kconfigs in my opinion than we currently do.

Reply:
Sure- the concern I have here is that the reason we use the pattern of `select MEMC` is to avoid corner cases. When a flash driver is enabled that requires the MEMC FlexSPI driver, the `MEMC_MCUX_FLEXSPI` driver must be enabled, so using `select` makes sense.

I understand that this is designed to fix the issue highlighted in #62788, but it will introduce new issues. What about a case where the `zephyr,flash` node is set to an SRAM region, but the user still wants to use the flash driver (for example for LittleFS)?

Reply:
> What about a case where the zephyr,flash node

what does this have to do with anything

>  is set to an SRAM region

why would a `zephyr,flash-controller` ever be set to an SRAM region

Reply:
> why would a zephyr,flash-controller ever be set to an SRAM region

I didn't see that you were using `zephyr,flash-controller`. Can you rename the `DT_CHOSEN_Z_FLASH` macro to `DT_CHOSEN_Z_FLASH_CONTROLLER`?

> What you just said using english is that a flash driver written to use the flexspi depends on MEMC_MCUX_FLEXSPI. select doesn't necessarily make sense.

`select` does makes sense in this context, imo. If the flash driver is written to use the MEMC FlexSPI API, it needs the MEMC FlexSPI driver to be enabled. Either it uses `select` to enable it, or it uses `depends on` and relies on another Kconfig snippet to enable `CONFIG_MEMC_MCUX_FLEXSPI`. Either way, the Kconfig must be enabled for the flash driver to function.

I still don't fully understand how this fixes the issue highlighted in https://github.com/zephyrproject-rtos/zephyr/pull/62788. In the current push, setting `CONFIG_MEMC=y` will now enable `CONFIG_MEMC_MCUX_FLEXSPI` by default (unless I'm missing something). This is the behavior we want to avoid if we are going to enable `CONFIG_MEMC` by default in every build

Reply:
`select MEMC` does not make sense, and our reason for using it previously and now the reason for this trickery that I am doing to work around the issue should not have to happen because the crux of the issue is that `CONFIG_MEMC` should not exist, and the only reason we are dealing with this nonsense is because we need to enable this meaningless config in the most roundabout way possible for the sake of... folder structure?

`select MEMC` inherently is a meaningless phrase because there is no such thing as a `MEMC` subsystem or a `MEMC` bus or even a `MEMC` API, the *only reason* for ever setting this KConfig to `y` is because some other driver `depends on` a **specific** driver in the `drivers/memc` folder, and only functions if that driver is built.... this is the purpose of `depends on`, and we need to describe the dependency of these flash drivers on the FLEXSPI driver, otherwise the description is either incomplete or incorrect. And we cannot select MEMC from the same config because this causes a dependency loop (but who cares because MEMC is meaningless and should not exist) The *actual logic* for why we want to turn it on is `(CONFIG_FLASH=y) AND (flash is controlled by flexspi) AND (xyz_flexspi flash driver is used)`. Therefore this version is explicit and clear about what is actually going on here and is more correct.

> I still don't fully understand how this fixes the issue highlighted in https://github.com/zephyrproject-rtos/zephyr/pull/62788. In the current push, setting CONFIG_MEMC=y will now enable CONFIG_MEMC_MCUX_FLEXSPI whenever the flexspi node has status = "okay" (unless I'm missing something). This is the behavior we want to avoid if we are going to enable CONFIG_MEMC by default in every build

The whole point is that now you can set flexspi to `status="disabled"` in DT and still use the flash node for XIP since the flash node being status okay doesn't `select` MEMC and the MEMC_MCUX_FLEXSPI erroneously.